src/components/dls/DurationInput/index.tsx

```diff
@@ -0,0 +1,93 @@
+import { ChangeEvent, useEffect, useState } from 'react';
+
+import classNames from 'classnames';
+import useTranslation from 'next-translate/useTranslation';
+
+import styles from './DurationInput.module.scss';
+
+import { convertNumberToDecimal } from '@/utils/number';
+
+interface DurationInputProps {
+  totalSeconds: number;
+  onTotalSecondsChange: (totalSeconds: number) => void;
+  disabled?: boolean;
+  label?: string;
+}
+
+const DurationInput = ({
+  totalSeconds,
+  onTotalSecondsChange,
+  disabled,
+  label,
+}: DurationInputProps) => {
+  const [hours, setHours] = useState<number>(0);
+  const [minutes, setMinutes] = useState<number>(0);
+  const [seconds, setSeconds] = useState<number>(0);
+  const { t } = useTranslation('common');
+
+  const handleChange = (setter: (value: number) => void) => (e: ChangeEvent<HTMLInputElement>) => {
+    const value = Number(e.target.value);
+    setter(value);
+  };
+
+  useEffect(() => {
+    onTotalSecondsChange(hours * 3600 + minutes * 60 + seconds);
+  }, [onTotalSecondsChange, hours, minutes, seconds]);
+
+  useEffect(() => {
+    const newHours = Math.floor(totalSeconds / 3600);
+    const newMinutes = Math.floor((totalSeconds % 3600) / 60);
+    const newSeconds = totalSeconds % 60;
+
+    setHours(convertNumberToDecimal(newHours, 1));
+    setMinutes(convertNumberToDecimal(newMinutes, 1));
+    setSeconds(convertNumberToDecimal(newSeconds, 1));
+  }, [totalSeconds]);
```

Comment:
This causes an infinite state update when inputs are below 0 using the mouse's scroll.
![Sep-03-2023 11-23-42](https://github.com/quran/quran.com-frontend-next/assets/15169499/87301e11-5bba-489f-885d-ced792b5fae3)